src/services/bindService.js

```diff
@@ -0,0 +1,64 @@
+'use strict';
+
+angular.module("openshiftCommonServices")
+  .service("BindingService", function($filter, $q, DataService, DNS1123_SUBDOMAIN_VALIDATION){
+    var bindingResource = {
+      group: 'servicecatalog.k8s.io',
+      resource: 'bindings'
+    };
+
+    var getProjectContext = function(project) {
+      return {
+        namespace: project.metadata.name
+      };
+    };
+
+    var makeBinding = function (serviceToBind) {
+      var generateName = $filter('generateName');
+
+      return {
+        kind: 'Binding',
+        apiVersion: 'servicecatalog.k8s.io/v1alpha1',
+        metadata: {
+          generateName: serviceToBind + '-'
+        },
+        spec: {
+          instanceRef: {
+            name: serviceToBind
+          },
+          secretName: generateName(_.trunc(serviceToBind, DNS1123_SUBDOMAIN_VALIDATION.maxlength - 6) + '-')
+        }
+      };
+    };
+
+    return {
+      bindService: function(project, serviceInstance, appToBind) {
+        var deferred = $q.defer();
+        var context = getProjectContext(project);
+        var newBinding = makeBinding(_.get(serviceInstance, 'metadata.name'));
+
+        // TODO: Use appToBind to bind the service to the application
+        DataService.create(bindingResource, null, newBinding, context).then(function(binding) {
+          deferred.resolve([binding, bindingResource, context]);
+        }, function(e) {
+          deferred.reject(e);
+        });
+
+        return deferred.promise;
+      },
+      bindApplication: function(project, appInstance, serviceToBind) {
+        var deferred = $q.defer();
+        var context = getProjectContext(project);
+        var newBinding = makeBinding(serviceToBind);
+
+        // TODO: Use the appInstance to bind the service to the application
+        DataService.create(bindingResource, null, newBinding, context).then(function(binding) {
+          deferred.resolve([binding, bindingResource, context]);
+        }, function(e) {
+          deferred.reject(e);
+        });
+
+        return deferred.promise;
```

Comment:
is there really any difference between bindService and bindApplication? you just need one method that creates a binding for a service and optionally binds it to an application (to support the just create a secret option)